File: projects/java/map2d/src/test/java/net/ponderingprogrammer/map2d/ConcreteFixedSquareMap2d.java
package net.ponderingprogrammer.map2d;

import net.ponderingprogrammer.map2d.Cell;
import net.ponderingprogrammer.map2d.CellGraph;
import net.ponderingprogrammer.map2d.FixedSquareMap2d;

import java.util.Collection;

/**
 * For testing FixedSquareMap2d
 */
public class ConcreteFixedSquareMap2d extends FixedSquareMap2d<Integer> {
    public ConcreteFixedSquareMap2d(int width, int height) {
        super(width, height);
    }

    @Override
    public Collection<Cell<Integer>> findAdjacentCells(int x, int y) {
        return null;
    }

    @Override
    public CellGraph<Integer> toCellGraph() {
        return null;
    }

}

File: projects/dotnet/NProcGen/tests/PonderingProgrammer.Map2d.Tests/ConcreteFixedSquareMap2d.cs
using System.Collections.Generic;

namespace PonderingProgrammer.Map2d.Tests
{
    class ConcreteFixedSquareMap2d : FixedSquareMap2d<bool>
    {
        public ConcreteFixedSquareMap2d(int width, int height) : base(width, height)
        {
        }

        public override IEnumerable<Cell<bool>> FindAdjacentCells(int x, int y)
        {
            return null;
        }

        public override CellGraph<bool> ToCellGraph()
        {
            return null;
        }
    }
}
